src/components/sbb-toggle/sbb-toggle.tsx

```diff
@@ -0,0 +1,200 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  JSX,
+  Listen,
+  Prop,
+  State,
+  Watch,
+} from '@stencil/core';
+import { InterfaceSbbToggleAttributes } from './sbb-toggle.custom';
+import { AgnosticResizeObserver as ResizeObserver } from '../../global/helpers/resize-observer';
+import getDocumentWritingMode from '../../global/helpers/get-document-writing-mode';
+
+let nextId = 0;
+
+/**
+ * @slot unnamed - Slot used to render the `<sbb-toggle-option>`.
+ */
+
+@Component({
+  shadow: true,
+  styleUrl: 'sbb-toggle.scss',
+  tag: 'sbb-toggle',
+})
+export class SbbToggle {
+  /**
+   * Id of the toggle element.
+   */
+  @Prop() public sbbToggleId = `sbb-toggle-${++nextId}`;
+
+  /**
+   * Whether the toggle is disabled.
+   */
+  @Prop() public disabled = false;
+
+  /**
+   * Set the width of the component.
+   */
+  @Prop() public even: boolean;
+
+  /**
+   * Size variant, either m or s.
+   */
+  @Prop() public size?: InterfaceSbbToggleAttributes['size'] = 'm';
+
+  /**
+   * The value of the toggle.
+   */
+  @Prop({ mutable: true, reflect: true }) public value: any | null;
+
+  /**
+   * Whether the list has an active action.
+   */
+  @State() private _hasCheckedOption = false;
+
+  @Element() private _element: HTMLElement;
+
+  private _toggleElement: HTMLElement;
+  private _toggleResizeObserver = new ResizeObserver(() => this._setCheckedPillPosition());
+
+  @Watch('value')
+  public valueChanged(value: any | undefined): void {
+    for (const toggleOption of this._options) {
+      toggleOption.checked = toggleOption.value === value;
+      toggleOption.tabIndex = this._getOptionTabIndex(toggleOption);
+    }
+    this._setFocusableOption();
+    this._setCheckedPillPosition();
+    this.didChange.emit({ value });
+  }
+
+  @Watch('disabled')
+  public updateDisabled(): void {
+    for (const toggleOption of this._options) {
+      toggleOption.disabled = toggleOption.disabled ? toggleOption.disabled : this.disabled;
+      toggleOption.tabIndex = this._getOptionTabIndex(toggleOption);
+    }
+    this._setFocusableOption();
+  }
+
+  /**
+   * Emits whenever the toggle value changes.
+   */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'did-change',
+  })
+  public didChange: EventEmitter<any>;
+
+  private get _options(): HTMLSbbToggleOptionElement[] {
+    return Array.from(
+      this._element.querySelectorAll('sbb-toggle-option')
+    ) as HTMLSbbToggleOptionElement[];
+  }
+
+  private get _checked(): HTMLSbbToggleOptionElement {
+    return this._options.find((toggle) => toggle.checked);
+  }
+
+  @Listen('did-select', { passive: true })
+  public onToggleOptionSelect(event: CustomEvent<Set<string>>): void {
+    this.value = event.detail;
+  }
+
+  private _setCheckedPillPosition(): void {
```

Comment:
resizing the window does not stable update the pill position, so I often get into situation where the pill is calculated at the wrong time (probably). Can we make this more reliable?

Reply:
See https://638f217768c512082108d28d--lyne-components-storybook.netlify.app/?path=/story/components-form-elements-toggle-sbb-toggle--dynamic-width

Reply:
Possible solution, maybe wait for another tick.